### PR TITLE
Implement a cross platform file monitoring beacon

### PR DESCRIPTION
**Pull request by wagdav**
### What does this PR do?

This PR implements a cross platform file monitoring beacon.  This is done by wrapping the watchdog Python library.

This PR was co-authored with @marco-m

### What issues does this PR fix or reference?

#28135

### New Behavior
A new watchdog beacon can be configured

### Tests written?

Yes

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by rares-pop** [comment]
Is this going to deprecate the inotify beacon?

**Comment by wagdav** [comment]
@rares-pop I don't think it should/could deprecate the inotify beacon, in my opinion. The `watchdog` library only exposes a small subset of inotify's features.

**Comment by wagdav** [comment]
I fixed all the pylint and codeclimate issues.  The remaining failing checks don't seem to be related to this branch.

**Review by garethgreenaway — APPROVED** [review]

**Comment by garethgreenaway** [comment]
@wagdav Very nice!

**Comment by marco-m** [comment]
Thanks for merging! SaltStack helped us greatly and this is a small give back.

File: salt/beacons/watchdog.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+'''
+watchdog beacon
```
Comment (rallytime) [inline salt/beacons/watchdog.py]: Can you add a `.. versionadded:: Fluorine` tag to this new beacon?

File: salt/beacons/watchdog.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+Watch files and translate the changes into salt events
+
+:depends:   - watchdog Python module >= 0.8.3
+
+'''
+# Import Python libs
+from __future__ import absolute_import
```
Comment (rallytime) [inline salt/beacons/watchdog.py]: `print_function` and `unicode_literals` needs to be included in the import line as well.

File: tests/unit/beacons/test_watchdog_beacon.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
+# coding: utf-8
+
+# Python libs
+from __future__ import absolute_import
```
Comment (rallytime) [inline tests/unit/beacons/test_watchdog_beacon.py]: Same import comment here as above.

**Review by rallytime — COMMENTED** [review]
Hi @wagdav - I have some small requests for this PR. Can you make those updates?

**Review by rallytime — APPROVED** [review]
You can squash them if you like, but it isn't necessary.

Thanks!

**Comment by wagdav** [comment]
@rallytime I squashed the fixes and rebased onto the develop branch.

**Pull request merged**